### PR TITLE
[Buttons] Refactor CSS vars 

### PR DESCRIPTION
**Pull request by jeremie-lucca**
## Description

Refactor CSS vars for buttons

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1931/storybook

File: packages/scss/src/components/button/component.scss [hunk 1]
```diff
@@ -19,14 +23,10 @@
 	color: var(--palettes-text, var(--palettes-primary-text));
 	cursor: pointer;
 
-	.lucca-icon {
+	.button-icon, .lucca-icon { // .lucca-icon legacy
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/button/component.scss:26]: where does this class come from ?

Comment (jeremie-lucca) [reply packages/scss/src/components/button/component.scss:26]: It used to be mandatory on `button mod-icon` but we slowly started to target `.lucca-icon`, which is a bad practice considering [our guidelines](https://prisme.lucca.io/94310e217/p/929c63-guidelines-intgration/t/30c6d8). So we reintroduced this class and considered `.lucca-icon` as deprecated.

Comment (AlexiaAlarza) [reply packages/scss/src/components/button/component.scss:26]: considering to [this guidelines](https://prisme.lucca.io/94310e217/p/929c63-guidelines-intgration/t/57f49d) it seems to be normal to have `.lucca-icon` and not a new class as it's en encapsulation component, isn't it ?

**Review by AlexiaAlarza — APPROVED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-1931/storybook

**Pull request merged**